crawl-ref/source/ng-wanderer.cc

```diff
@@ -747,3 +747,24 @@ void create_wanderer()
 
     _wanderer_cover_equip_holes();
 }
+
+void memorise_wanderer_spell()
+{
+    // If the player got only one level 1 spell, memorise it. Otherwise, let the
+    // player choose which spell(s) to memorise and don't memorise any.
+    auto const available_spells = get_sorted_spell_list(true, true);
+    if (available_spells.size())
+    {
+        int num_level_one_spells;
```

Comment:
It looks like the problem I was having with spells not memorising is this being uninitialised, initialising it at 0 makes it work as expected for me.

Reply:
Hmm that's weird, does this vary between compilers?

Reply:
yes, it definitely can; and specificaly clang often behaves differently than gcc w.r.t. uninitialized variables.

Reply:
That probably explains why mine worked fine then, thanks.